src/nvim/ex_cmds2.c

```diff
@@ -3828,7 +3828,13 @@ static void script_host_execute(char *name, exarg_T *eap)
     // current range
     tv_list_append_number(args, (int)eap->line1);
     tv_list_append_number(args, (int)eap->line2);
-    (void)eval_call_provider(name, "execute", args);
+
+    if (!eval_has_provider(name)) {
+        EMSG("No provider found. Run \":checkhealth provider\" for more "
+            "information.");
```

Comment:
"Run foo" is enough, don't need "for more information".

Reply:
The message should include the `name` value, to give more context.

    emsgf("No \"%s\" provider found. Run \":checkhealth provider\", name);

Reply:
It could be a good idea to print the provider name in the message. `Provider for NAME not found. Run :checkhealth`